packages/features/tasker/tasks/triggerFormSubmittedNoEvent/formSubmissionValidation.ts

```diff
@@ -0,0 +1,122 @@
+import prisma from "@calcom/prisma";
+
+export interface ValidationOptions {
+  responseId: number;
+  formId: string;
+  responses: any;
+  submittedAt?: Date;
+}
+
+export interface ValidationResult {
+  skip: boolean;
+  reason?: string;
+}
+
+/**
+ * Check if trigger should be skipped due to booking creation or duplicate submission
+ */
+export async function shouldTriggerFormSubmittedNoEvent(options: ValidationOptions) {
+  const { formId, responseId, responses, submittedAt } = options;
+
+  // Check if a booking was created from this form response
+  const bookingExists = await hasBooking(responseId);
+
+  if (bookingExists) return false;
+
+  // Check for duplicate form submissions
+  const hasDuplicate = await hasDuplicateSubmission({ formId, responseId, responses, submittedAt });
+  if (hasDuplicate) {
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * Check if a booking was created from this form response
+ */
+async function hasBooking(responseId: number): Promise<boolean> {
+  const bookingFromResponse = await prisma.booking.findFirst({
+    where: {
+      routedFromRoutingFormReponse: {
+        id: responseId,
+      },
+    },
+    select: {
+      id: true,
+    },
+  });
+
+  return !!bookingFromResponse;
+}
+
+export function getSubmitterEmail(responses: any) {
+  const submitterEmail = Object.values(responses).find(
+    (response): response is { value: string; label: string } => {
+      const value =
+        typeof response === "object" && response && "value" in response ? response.value : response;
+      return typeof value === "string" && value.includes("@");
```

Comment:
Can we use better email validation here? Also what would happen in case of phone based bookings?

Reply:
That's the same logic that we used before in `triggerFormSubmittedNoEventWebhook.ts`. Now we also reuse it for finding the email address in the form submission for the `EMAIL_ATTENDEE` workflow action. This workflow action won't work for phone based bookings which is fine